vignettes/hys_law.Rmd

```diff
@@ -0,0 +1,145 @@
+- No other reason to explain these increased lab values like preexisting liver disease.
+
+
+# Example
+We can use a mixture of `admiral` and basic `tidyverse` functions to assist us in creating an ADHY dataset. In the walkthrough below we will use the adlb dataset created from the call `use_ad_template("adlb")`
+
+```{r}
+adlb <- admiral_adlb
+head(adlb)
+```
+
+Using `call_derivation()` and `derive_var_merged_exist_flag()`, we can create a variety of flags such as one to indicate the the 3-fold or greater than upper limit of normal of ALT/AST. The categorical bands and equality symbol placement may vary with your organization. In our example, for ALT/AST we'll create 4 categories: <3xULN, >=3 - <5xULN, >=5 - <10xULN, >=10xULN.
+
+```{r}
+adlb2 <- call_derivation(
+  adlb,
+  derivation = derive_var_merged_exist_flag,
+  dataset_add = adlb,
+  filter_add = PARAMCD %in% c("AST", "ALT", "BILI") & is.na(DTYPE),
+  by_vars = vars(USUBJID, LBSEQ, PARAMCD, ADT),
+  variable_params = list(
+    params(
+      new_var = AST01FL,
+      condition = AVAL / ANRHI < 3 & PARAMCD %in% c("AST", "ALT")
+    ),
+    params(
+      new_var = AST02FL,
+      condition = 3 <= AVAL / ANRHI & AVAL / ANRHI < 5 & PARAMCD %in% c("AST", "ALT")
+    ),
+    params(
+      new_var = AST03FL,
+      condition = 5 <= AVAL / ANRHI & AVAL / ANRHI < 10 & PARAMCD %in% c("AST", "ALT")
+    ),
+    params(
+      new_var = AST04FL,
+      condition = 10 <= AVAL / ANRHI & PARAMCD %in% c("AST", "ALT")
+    ),
+    params(
+      new_var = BILI01FL,
+      condition = AVAL / ANRHI < 2 & PARAMCD %in% c("BILI")
+    ),
+    params(
+      new_var = BILI02FL,
+      condition = 2 <= AVAL / ANRHI & PARAMCD %in% c("BILI")
+    )
+  )
+) %>% 
+  filter(PARAMCD %in% c("AST", "ALT", "BILI"))
+head(adlb2)
+```
+
+Filter down the instances that you may want to include in your DILI analysis using the flags created above. By doing a pivot from wide to long you end up with a dataset that will be easier to join on itself to find all the paired instances of AST/ALT & BILI within a certain time-window.
+
+```{r}
+adlb3 <- adlb2 %>%
+  filter(is.na(DTYPE)) %>%
+  filter(AST02FL == "Y" | AST03FL == "Y" | AST04FL == "Y" | BILI02FL == "Y")%>%
+  select(USUBJID, LBSEQ, PARAMCD, ADY, ADT, starts_with(c("AST", "BILI"))) %>%
+  pivot_longer(.,
+    starts_with(c("AST", "BILI")),
+    names_to = "DILICRIT",
+    values_to = "POTDILI"
+  ) %>%
+  #filter(!is.na(POTDILI)) %>%
+  arrange(ADY) %>%
+  mutate(dummy = 1)
+head(adlb3)
+```
+
+Splitting the dataset into its AST/ALT subet and BILI subset, then joining these two datasets, the resulting dataset is one that contains each AST/ALT lab record cross referenced with each BILI lab record. Filtering records that occured within a time window, for example 14 days, then selecting on distinct records of `USUBJID` and `ADT` you can identify paired records that may qualify for DILI. 
```

Comment:
@zdz2101 , did you look at `derive_var_confirmation_flag()` and `derive_vars_joined()`? The code looks similar to what these functions are doing. Do we need something like a `derive_param_joined()` function?